demo/app.js

```diff
@@ -77,12 +77,15 @@ class App extends React.Component {
           <div>
             <button
               onClick={() =>
-                this.setState({
-                  length: 2
+                this.setState(prevState => {
+                  const length = prevState.length === 6 ? 2 : 6;
+                  return {
+                    length
+                  };
                 })
               }
             >
-              2 Slides Only
+              Toggle 2 Slides
```

Comment:
maybe rename this to `Toggle Show 2 Slides Only` - it's a little unclear what `2 slides` could mean to a new user of the lib